src/axom/primal/operators/intersect.hpp

```diff
@@ -299,7 +306,49 @@ bool intersect(const OrientedBoundingBox< T, 3 >& b1,
   return detail::intersect_obb3D_obb3D(b1, b2, EPS);
 }
 
+
+/*!
+ * \brief Tests if two Bezier Curves \a c1 and \a c2 intersect.
+ * \return status true iff \a c1 intersects \a c2, otherwise false.
+ *
+ * \param [in] c1 the first BezierCurve, parametrized in [0,1)
+ * \param [in] c2 the second BezierCurve, parametrized in [0,1)
+ * \param [out] sp vector of parameter space intersection points for \a c1
+ * \param [out] tp vector of parameter space intersection points for \a c2
+ * \param [in] tol tolerance parameter for determining if a curve can
+ * be approximated by a line segment.
+ * \return True if the curves intersect, false otherwise. Intersection
+ * parameters are stored in \a sp and \a tp
+ *
+ * Finds all intersection points between the two curves.
+ *
+ * \note This function assumes that the curves are in general position.
+ * Specifically, we assume that all intersections are at points and that
+ * the curves don't overlap.
+ *
+ * \note This function assumes that the curves are half-open, i.e. they
+ * contain their first endpoint, but not their last endpoint. Thus, the
+ * curves do not intersect at \f$ s==1 \f$ or at \f$ t==1 \f$.
```

Comment:
This PR does not handle the case where the curves intersect at their last endpoint.  
We will address this in a forthcoming PR.